src/Serilog.Settings.Configuration/Settings/Configuration/Assemblies/DllScanningAssemblyFinder.cs

```diff
@@ -58,4 +33,34 @@ static AssemblyName TryGetAssemblyNameFrom(string path)
             }
         }
     }
+
+    static IEnumerable<string> GetProbeDirs()
+    {
+        if (!string.IsNullOrEmpty(AppDomain.CurrentDomain.BaseDirectory))
```

Comment:
nit: i would invert this to remove the negation and do an early exit to reduce nesting

Reply:
I have now rewritten this pull request without touching the `DllScanningAssemblyFinder` implementation, except to remove the unnecessary `AsReadOnly()` call.